src/test_utils/public/http_test_setup.ts

```diff
@@ -19,20 +19,29 @@
 
 /* eslint-disable @kbn/eslint/no-restricted-paths */
 import { HttpService } from '../../core/public/http';
-import { BasePathService } from '../../core/public/base_path';
 import { fatalErrorsServiceMock } from '../../core/public/fatal_errors/fatal_errors_service.mock';
 import { injectedMetadataServiceMock } from '../../core/public/injected_metadata/injected_metadata_service.mock';
 /* eslint-enable @kbn/eslint/no-restricted-paths */
 
-export function setup() {
-  const httpService = new HttpService();
-  const fatalErrors = fatalErrorsServiceMock.createSetupContract();
-  const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+export type SetupTap = (
+  injectedMetadata: ReturnType<typeof injectedMetadataServiceMock.createSetupContract>,
+  fatalErrors: ReturnType<typeof fatalErrorsServiceMock.createSetupContract>
+) => void;
 
+const defaultTap: SetupTap = (
+  injectedMetadata: ReturnType<typeof injectedMetadataServiceMock.createSetupContract>
+) => {
   injectedMetadata.getBasePath.mockReturnValue('http://localhost/myBase');
+};
+
+export function setup(tap: SetupTap = defaultTap) {
+  const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+  const fatalErrors = fatalErrorsServiceMock.createSetupContract();
 
-  const basePath = new BasePathService().setup({ injectedMetadata });
-  const http = httpService.setup({ basePath, fatalErrors, injectedMetadata });
+  tap(injectedMetadata, fatalErrors);
+
+  const httpService = new HttpService();
```

Comment:
question: why we create an instance of a real service here, but instances of mocked services above?